### PR TITLE
Move extension templates to tool folder

### PR DESCRIPTION
**Pull request by davidkline-ms**
## Overview
The template files used by the ExtensionServiceCreator tool were previously located in the root of the project (peer to the Assets folder). This location is not valid as they cannot be navigated to by the user (from within Unity) and prevents them from being packaged for distribution.

This change moves them into a Templates folder beneath MixedRealityToolkit.Tools/ExtensionServiceCreator and updates the paths within the tool.

## Validation
Confirmed that the tool successfully finds and loads the tempates.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs [hunk 1]
```diff
@@ -68,10 +68,10 @@ private struct PersistentState
         private static readonly string PersistentStateKey = "MRTK_ExtensionServiceWizard_State_Before_Recompilation";
         private static readonly string DefaultExtensionsFolder = "Assets/MixedRealityToolkit.Extensions";
         private static readonly string DefaultExtensionsFolderName = "MixedRealityToolkit.Extensions";
-        private static readonly string ServiceTemplatePath = "ExtensionTemplates/ExtensionScriptTemplate.txt";
-        private static readonly string InspectorTemplatePath = "ExtensionTemplates/ExtensionInspectorTemplate.txt";
-        private static readonly string InterfaceTemplatePath = "ExtensionTemplates/ExtensionInterfaceTemplate.txt";
-        private static readonly string ProfileTemplatePath = "ExtensionTemplates/ExtensionProfileTemplate.txt";
+        private static readonly string ServiceTemplatePath = "Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/Templates/ExtensionScriptTemplate.txt";
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: Does this path resolve properly even when consumed via Nuget?

File: Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs [hunk 2]
```diff
@@ -68,10 +68,10 @@ private struct PersistentState
...
+        private static readonly string InspectorTemplatePath = "Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/Templates/ExtensionInspectorTemplate.txt";
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: Also, I think there was some work done previously to support the MRTK in different locations / folder names than the default

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: This file I think https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs

Comment (wiwei) [reply Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: Yep - look up MixedRealityToolkitFiles.cs - has some stuff around identifying where things are located at

File: Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: I will need to work with @andreiborodin on this part. I do not believe they are consumable via NuGet now.

File: Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: Looks like we need to add Extensions and Tools to that list.

As a note, this was a simple string replace. None of this class leverages the toolkit files class.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: updated the tool source. thanks!

File: Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: talked with @andreiborodin and it looks like these files will be packaged correctly.

File: Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs [hunk 3]
```diff
@@ -85,6 +81,38 @@ private struct PersistentState
 
         #endregion
 
+        #region paths
+
+        private string ExtensionsFolder
+        {
+            get
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: nit: inconsistent syntax between this (`get { }`) and the ones following (`get =>`).

Could we turn them all into `private string Foo => MixedRealityToolkitFiles.Bar()`? :)

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: i'll update the syntax :)

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Tools/ExtensionServiceCreator/ExtensionServiceCreator.cs]: fixed

**Pull request merged**